Backtracking/GenerateSubsets

```diff
@@ -0,0 +1,26 @@
+function generateSubsets(nums) {
+    const subsets = [];
+    
+    function backtrack(start = 0, currentSubset = []) {
```

Comment:
I'm not sure I would call this backtracking; to me it's just ordinary recursion:

> Backtracking is a class of algorithms for finding solutions to some computational problems, notably constraint satisfaction problems, that incrementally builds candidates to the solutions, and abandons a candidate as soon as it determines that the candidate cannot possibly be completed to a valid solution.